src/utils/string.hpp

```diff
@@ -459,4 +459,30 @@ inline std::string_view Substr(const std::string_view string, size_t pos = 0, si
   return string.substr(pos, len);
 }
 
+/**
+ * Convert a double value to a string representation.
+ * Precision of converted value is 16.
+ * Function also removes trailing zeros after the dot.
+ *
+ * @param value The double value to be converted.
+ *
+ * @return The string representation of the double value.
+ *
+ * @throws None
+ */
+inline std::string DoubleToString(const double value) {
+  static const int PRECISION = 16;
+  static const int STR_LENGTH = PRECISION + 1;  // 1 for dot
+
+  char buf[32];
```

Comment:
Using `stringstream` would be more c++-like.
Plus you could avoid a copy between buf and string and at the end move the stringstream, so we make sure to avoid that copy as well. 
See `std::precision` and `std::fixed`. It's not super powerful, it almost does what you want, but there are some issue with rounding and scientific notation used in some cases. So, I think, at the end you will need to do the same manipulation you are currently doing.

Reply:
I thought about `stringstream`. Yes, it's more C++ way. I will test this approach.
I used `sprintf` because it's under the hood of `std::to_string`

Reply:
`fmt::format("{:.16f}", d)` https://godbolt.org/z/ehGYcco5d
But note adding precision inn cases like 5.8 it makes the number returned worse.

Reply:
For this reason I prefer to use `sprintf` or `stringstream`.

Reply:
My point was that that precision of 16, in all cases, will make make `5.8` -> `5.7999999999999998` no matter what. This is because the actual represented value is `5.79999999999999982236431605997495353221893310546875` and hence with 16 digits it rounds the boundary 82 down to 80 hence `5.7999999999999998`. If we used 15 digits then boundary 98 would round up to 100 giving us clean `5.800000000000000`.

Reply:
I'm not sure what is generally the best practice for printing floating numbers.
All examples https://godbolt.org/z/8sPbbenxW

Reply:
Look into https://github.com/ulfjack/ryu 
> For example, consider the binary 32-bit floating point number 00111110100110011001100110011010. The stored value is exactly 0.300000011920928955078125. However, this floating point number is also the closest number to the decimal number 0.3, so that is what Ryu outputs.